param_persist/agents/sqlalchemy_agent.py

```diff
@@ -59,18 +60,27 @@ def save(self, instance, **kwargs):
             The id of the row in the database corresponding to the parameterized instance.
         """
         db_session = kwargs.get('db_session', None)
-        serialized_param_dict = ParamSerializer.to_dict(instance)
+
+        # Serialize data using param JSONSerialization class
+        serialized_param = json.loads(JSONSerialization.serialize_parameters(instance))
+
+        # Remove name since we don't need it
+        serialized_param.pop('name')
+
+        # Get class path and uuid to save in InstanceModel
+        class_path = self.get_class_path_from_param_instance(instance)
         new_instance_uuid = uuid.uuid4()
-        new_instance = InstanceModel(id=str(new_instance_uuid), class_path=serialized_param_dict.get('class_path'))
+        new_instance = InstanceModel(id=str(new_instance_uuid), class_path=class_path)
         db_session.add(new_instance)
 
-        param_models = [ParamModel(id=(str(uuid.uuid4())), value=json.dumps(param),
+        param_models = [ParamModel(id=(str(uuid.uuid4())),
```

Comment:
I appreciate the use of dict comprehension here, but when it gets longer than a line or two it becomes confusing. Please change this and other long dict or list comprehensions to regular for loops.